### PR TITLE
Solved issue #411

### PR DESCRIPTION
**Pull request by sarthak-g**
Added fullscreen and unfullscreen button in Gears Activity as mentioned in issue #411 

gear1

gear2

### CODE REVIEW

**Comment by llaske** [comment]
That's the idea.
But there is an error in the console when clicking on fullscreen/unfullscreen button:

```
activity.js:126 Uncaught ReferenceError: resizeHandler is not defined
    at HTMLButtonElement.<anonymous> (activity.js:126)
```

**Comment by sarthak-g** [comment]
I've updated the code and made a commit.

**Comment by llaske** [comment]
There is another issue: in fullscreen mode, the drawing is not aligned with the cursor. There is vertical shift to the size of the toolbar.

image

**Comment by sarthak-g** [comment]
If vertical shift will not be done then blank space will display as below:
VerticalShiftNotDone
@canvasOffsetY in gearsketch_main.coffee gets value only when the page is displayed for first time (with toolbar), so when fullscreen is done then size of the canvas is not updated.
 What I think is we've to call @updateCanvasSize() when fullscreen as well as unfullscreen button is pressed but for doing so we need jquery.

**Comment by llaske** [comment]
Of course the vertical shift should be done but it should be take into account when mouse position is computed.
You're right, I guess the issue is related to the canvasOffsetY initialized [here](https://github.com/llaske/sugarizer/blob/dev/activities/Gears.activity/lib/gearsketch/gearsketch_main.js#L92). This value should be updated each time there is a click on fullscreen/unfullscreen button. 
Calling updateCanvasSize method of the gearSketch object, from the activity, could be a solution. I don't think you need jQuery since the gearSketch object is available from the activity.

**Comment by sarthak-g** [comment]
Fullscreen is working as expected.

File: activities/Gears.activity/js/activity.js [hunk 1]
```diff
@@ -117,6 +117,22 @@ define(["sugar-web/activity/activity","sugar-web/graphics/radiobuttonsgroup","ge
             gearSketch.playDemo();
         });
 
+        // Full screen
+		document.getElementById("fullscreen-button").addEventListener('click', function() {
```
Comment (llaske) [inline activities/Gears.activity/js/activity.js]: Use space instead of TAB as identation

File: activities/Gears.activity/js/activity.js [hunk 2]
```diff
@@ -117,6 +117,22 @@ define(["sugar-web/activity/activity","sugar-web/graphics/radiobuttonsgroup","ge
...
+			document.getElementById("main-toolbar").style.opacity = 0;
```
Comment (llaske) [inline activities/Gears.activity/js/activity.js]: Use space instead of TAB as identation

File: activities/Gears.activity/js/activity.js [hunk 3]
```diff
@@ -117,6 +117,22 @@ define(["sugar-web/activity/activity","sugar-web/graphics/radiobuttonsgroup","ge
...
+            document.getElementById("canvas").style.top = "0px";
+            gearSketch.canvasOffsetY = gearSketch.canvas.getBoundingClientRect().top;
+            gearSketch.updateCanvasSize();
+            document.getElementById("unfullscreen-button").style.visibility = "visible";
+		});
+		document.getElementById("unfullscreen-button").addEventListener('click', function() {
```
Comment (llaske) [inline activities/Gears.activity/js/activity.js]: Use space instead of TAB as identation

File: activities/Gears.activity/js/activity.js [hunk 4]
```diff
@@ -117,6 +117,22 @@ define(["sugar-web/activity/activity","sugar-web/graphics/radiobuttonsgroup","ge
...
+			document.getElementById("main-toolbar").style.opacity = 1;
```
Comment (llaske) [inline activities/Gears.activity/js/activity.js]: Use space instead of TAB as identation

**Comment by sarthak-g** [comment]
Indentation issue resolved.

**Comment by llaske** [comment]
Sorry, there is still a TAB instead of white spaces before lines `document.getElementById("fullscreen-button")`,  `});` and  `document.getElementById("unfullscreen-button")`

See the screen capture of Atom editor below:

image

**Comment by sarthak-g** [comment]
It is fine now I guess.
indent show spaces

**Comment by llaske** [comment]
Cool. That's fine now.
Thanks.

**Pull request merged**